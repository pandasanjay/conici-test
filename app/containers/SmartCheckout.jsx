import React, { Component } from "react";
import { connect } from "react-redux";
import { billingAddressAction, selectAddressAction } from "../actions";
import "./SmartCheckout.scss";
import ComponentLoader from "../components/ComponentLoader";
import BillingAddress from "../components/BillingAddressSection";

export class SmartCheckout extends Component {
  constructor(props) {
    super(props);
  }
  componentDidMount() {
    this.props.getBillingAddress();
  }
  render() {
    const { heading, description, components } =
      (this.props.globalReducer.pageContents &&
        this.props.globalReducer.pageContents.content) ||
      {};

    const { back, submit } =
      (this.props.globalReducer.pageContents &&
        this.props.globalReducer.pageContents.global) ||
      {};

    const addressList = this.props.billingAddressReducer.address || [];

    return (
      <div>
        <h2 className="text-center pt-3">{heading}</h2>
        <p className="text-center">{description}</p>
        {addressList.map((data, i) => (
          <BillingAddress
            {...data}
            key={i}
            id={i}
            onSelect={this.props.selectAddress}
          />
        ))}
        <ComponentLoader name="add_new_address" data={components || {}} />
        <div className="sh-actions d-flex mt-5 justify-content-between pt-2 pb-2">
          <div className="pt-2 pb-2">
            <button
              type="button"
              className="btn btn-outline-primary pl-5 pr-5 text-uppercase"
            >
              {back}
            </button>
          </div>
          <div className="pt-2 pb-2">
            <button
              type="button"
              className="btn btn-primary pl-5 pr-5 text-uppercase"
            >
              {submit}
            </button>
          </div>
        </div>
      </div>
    );
  }
}
/*
 * mapDispatchToProps
 */
const mapDispatchToProps = dispatch => ({
  getBillingAddress: () => dispatch(billingAddressAction()),
  selectAddress: id => dispatch(selectAddressAction(id))
});
/*
 * mapStateToProps
 */
const mapStateToProps = state => ({
  ...state
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SmartCheckout);
