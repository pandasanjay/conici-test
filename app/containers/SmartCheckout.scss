@import "../var.scss";

.sh-actions {
    border-top: 3px solid $secondry-color
}

.sh-add-card {
    cursor: pointer;
    border: 1px solid #000;
    background-color: #005578;
    color: #fff;
    border-radius: .5rem;
}

.sh-h3 {
    font-size: 24px;
    font-weight: bold;
}