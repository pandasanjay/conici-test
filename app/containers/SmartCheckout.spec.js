import React from "react";
import { SmartCheckout } from "./SmartCheckout";
import { shallow } from "enzyme";
describe("Smart checkout component with out data", () => {
  let component;
  beforeEach(() => {
    component = shallow(
      <SmartCheckout
        globalReducer={{}}
        getBillingAddress={jest.fn()}
        billingAddressReducer={{}}
      />
    );
  });
  it("should have match with snapshot", () => {
    expect(component).toMatchSnapshot();
  });
  it("should have property", () => {
    expect(component.instance().props.globalReducer).toEqual({});
    expect(component.instance().props.getBillingAddress).toBeCalled();
  });
});
const globalReducerStub = {
  pageContents: {
    content: {
      heading: "some data",
      description: "Some desc"
    },
    global: {
      back: "back",
      submit: "Submit"
    }
  }
};
describe("Smart checkout component with data", () => {
  let component;
  beforeEach(() => {
    component = shallow(
      <SmartCheckout
        globalReducer={globalReducerStub}
        getBillingAddress={jest.fn()}
        billingAddressReducer={{}}
      />
    );
  });
  it("should have match with snapshot", () => {
    expect(component).toMatchSnapshot();
  });
  it("should have property", () => {
    expect(component.instance().props.globalReducer).toEqual(globalReducerStub);
  });
});
