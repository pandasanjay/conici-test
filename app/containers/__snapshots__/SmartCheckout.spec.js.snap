// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smart checkout component with data should have match with snapshot 1`] = `
<div>
  <h2
    className="text-center pt-3"
  >
    some data
  </h2>
  <p
    className="text-center"
  >
    Some desc
  </p>
  <div
    className="sh-add-card d-flex pt-5 pb-5 p-3 justify-content-between"
  >
    <div>
      <div
        className="sh-h3"
      >
        ADD NEW
      </div>
      <div
        className="text-uppercase"
      >
        BILLING ADDRESS
      </div>
    </div>
    <div
      className="pt-4"
    >
      <Icon
        iconname="icon-add-active"
      />
    </div>
  </div>
  <div
    className="sh-actions d-flex mt-5 justify-content-between pt-2 pb-2"
  >
    <div
      className="pt-2 pb-2"
    >
      <button
        className="btn btn-outline-primary pl-5 pr-5 text-uppercase"
        type="button"
      >
        back
      </button>
    </div>
    <div
      className="pt-2 pb-2"
    >
      <button
        className="btn btn-primary pl-5 pr-5 text-uppercase"
        type="button"
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;

exports[`Smart checkout component with out data should have match with snapshot 1`] = `
<div>
  <h2
    className="text-center pt-3"
  />
  <p
    className="text-center"
  />
  <div
    className="sh-add-card d-flex pt-5 pb-5 p-3 justify-content-between"
  >
    <div>
      <div
        className="sh-h3"
      >
        ADD NEW
      </div>
      <div
        className="text-uppercase"
      >
        BILLING ADDRESS
      </div>
    </div>
    <div
      className="pt-4"
    >
      <Icon
        iconname="icon-add-active"
      />
    </div>
  </div>
  <div
    className="sh-actions d-flex mt-5 justify-content-between pt-2 pb-2"
  >
    <div
      className="pt-2 pb-2"
    >
      <button
        className="btn btn-outline-primary pl-5 pr-5 text-uppercase"
        type="button"
      />
    </div>
    <div
      className="pt-2 pb-2"
    >
      <button
        className="btn btn-primary pl-5 pr-5 text-uppercase"
        type="button"
      />
    </div>
  </div>
</div>
`;
